src/hangar/checkout.py

```diff
@@ -127,6 +134,236 @@ def __verify_checkout_alive(self):
+
+        except_missing: bool, **KWARG ONLY
+
+            If False, will not throw exceptions on missing sample key value.
+            Will raise KeyError if True and missing key found.
+
+        Returns
+        -------
+        Arrayset
+            single arrayset parameter, no samples specified
+
+        np.ndarray
+            Single arrayset specified, single sample key specified
+
+        List[np.ndarray]
+            Single arrayset, multiple samples array data for each sample is
+            returned in same order sample keys are recieved.
+
+        List[NamedTuple[*np.ndarray]]
+            Multiple arraysets, multiple samples. Each arrayset's name is used
+            as a field in the NamedTuple elements, each NamedTuple contains
+            arrays stored in each arrayset via a common sample key. Each sample
+            key is returned values as an individual element in the
+            List. The sample order is returned in the same order it wasw recieved.
+
+        """
+        try:
+            tmpconman = not self._is_conman
+            if tmpconman:
+                self.__verify_checkout_alive()
+                self.__enter__()
+
+            # Arrayset Parsing
+            if (arraysets is (Ellipsis, slice)):
+                arraysets = list(self._arraysets._arraysets.values())
+            elif isinstance(arraysets, str):
+                arraysets = [self._arraysets._arraysets[arraysets]]
+            elif isinstance(arraysets, (tuple, list)):
+                arraysets = [self._arraysets._arraysets[aname] for aname in arraysets]
+            else:
+                raise TypeError(f'Arraysets: {arraysets} type: {type(arraysets)}')
+            nAsets = len(arraysets)
+            ArraysetData = namedtuple('ArraysetData', [aset.name for aset in arraysets])
+
+            # Sample Parsing
+            if isinstance(samples, (str, int)):
+                samples = [samples]
+            elif not isinstance(samples, (tuple, list)):
+                raise TypeError(f'Samples idx: {samples} type: {type(samples)}')
+            nSamples = len(samples)
+
+            # Data Retrieval
+            asetsSamplesData = []
+            for aset in arraysets:
+                aset_samples = []
+                for sample in samples:
+                    try:
+                        arr = aset.get(sample)
+                    except KeyError as e:
+                        if except_missing:
+                            raise e
+                        arr = None
+                    aset_samples.append(arr)
+                if nAsets == 1:
+                    asetsSamplesData = aset_samples
+                    if nSamples == 1:
+                        asetsSamplesData = asetsSamplesData[0]
+                    break
+                asetsSamplesData.append(aset_samples)
+            else:  # N.B. for-else conditional (ie. 'no break')
+                tmp = map(ArraysetData._make, zip(*asetsSamplesData))
```

Comment:
thought: list comprehension instead of map and then list?

Reply:
I'll have to check on this one. `map` generates a loop which executes in the compiled `C-python` layer. I tend to use it whenever possible since it offers a decent speedup for long running loops. Not sure if we would see much of a difference here though if we used standard list comp. I'll let ya know.